ipalib/install/kinit.py

```diff
@@ -95,3 +95,34 @@ def kinit_password(principal, password, ccache_name, config=None,
                  capture_error=True)
     if result.returncode:
         raise RuntimeError(result.error_output)
+
+
+def kinit_armor(ccache_name):
+    """
+    perform kinit to obtain anonymous ticket to be used as armor for FAST.
+    """
+    root_logger.debug("Initializing anonymous ccache")
+
+    env = {'LC_ALL': 'C'}
+    # try with the keytab first and then again fallback to try with pkinit in
+    # case someone decided it is fun to remove Anonymous keys from the entry
+    # or in future pkinit enabled principal enforce the use of pkinit
+    try:
+        # Gssapi does not understand anonymous cred use kinit command instead
+        args = [paths.KINIT, '-k', '-t', paths.ANON_KEYTAB,
+                ANON_USER, '-c', ccache_name]
+        result = run(args, env=env, raiseonerr=False, capture_error=True)
+        if result.returncode:
```

Comment:
why returncode is always 0 here?  If returncode is 0, then no CalledProcessError is raised, because `run()` raises this exception only if returncode != 0